app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,28 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        self.ldap_con.valid_user?("")
+      end
+      result[:success] = true
+      result[:message] = _("Test connection to LDAP server was successful.")
+    rescue Timeout::Error, Net::LDAP::Error  => exception
+      exception = Foreman::Exception.new exception.message
+      result[:success] = false
+      result[:error_class] = exception.class.name
+      result[:message] = _(exception.message)
```

Comment:
i didnt mean the error code, currently the response is something like:
```ERFXX-NNNN [Foreman::Exception]```

my suggestion was to remove the string [Foreman::Exception], not the error code.